provider/workspace_owner.go

```diff
@@ -53,6 +53,12 @@ func workspaceOwnerDataSource() *schema.Resource {
 			_ = rd.Set("session_token", os.Getenv("CODER_WORKSPACE_OWNER_SESSION_TOKEN"))
 			_ = rd.Set("oidc_access_token", os.Getenv("CODER_WORKSPACE_OWNER_OIDC_ACCESS_TOKEN"))
 
+			if login_type := os.Getenv("CODER_WORKSPACE_OWNER_LOGIN_TYPE"); login_type != "" {
+				_ = rd.Set("login_type", login_type)
+			} else {
+				_ = rd.Set("login_type", "none")
```

Comment:
@mtojek should we instead set `login_type` to an empty string if Coder does not set the required env?

Reply:
Perhaps at the very least, drop a `diag.Warn`?

Reply:
> should we instead set login_type to an empty string

This seems reasonable and consistent with other ENV vars. I would start with empty string, we can always revisit it in the future and replace it with `none`.

Reply:
> This seems reasonable and consistent with other ENV vars. I would start with empty string, we can always revisit it in the future and replace it with `none`.

Great, what about the warn?

Reply:
Sorry, but taking a look the the `diag` package there is not `Warn` method.

https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/diag

Reply:
Sorry! I should have been more specific -- you can append a `diag.Diagnostic` with severity `diag.Warning` to the second return argument of type `diag.Diagnostics`.

Reply:
Done, this is the first time I have to deal with the `diags`, so please let me know if this solution is correct 🙏🏼 .